server/ocsp.go

```diff
@@ -558,10 +558,11 @@ func (s *Server) configureOCSP() []*tlsConfigKind {
 			tlsConfig: config,
 			tlsOpts:   opts,
 			apply: func(tc *tls.Config) {
-
 				// RequireAndVerifyClientCert is used to tell a client that it
 				// should send the client cert to the server.
-				tc.ClientAuth = tls.RequireAndVerifyClientCert
+				if opts.Verify {
+					tc.ClientAuth = tls.RequireAndVerifyClientCert
+				}
 				// GetClientCertificate is used by a client to send the client cert
 				// to a server. We're a server, so we must not set this.
 				tc.GetClientCertificate = nil
```

Comment:
Comment is confusing.  We are the server (leaf hub) in this case.  Does it mean that we would only use `tc.GetClientCertificate` to obtain the client's certificate just used in the successful TLS handshake?

If we as Leaf Hub are configured for `verify_and_map: true` then post-handshake we need to evaluate cert attributes for user mapping purposes.   Setting `nil` here defeat that function?

Reply:
I simplified the comment and also added a `verify_and_map` test to confirm.